src/zenml/integrations/skypilot/flavors/skypilot_orchestrator_aws_vm_flavor.py

```diff
@@ -0,0 +1,134 @@
+from zenml.integrations.skypilot.flavors.skypilot_orchestrator_base_vm_config import (
+    SkypilotBaseOrchestratorConfig,
+    SkypilotBaseOrchestratorSettings,
+)
+from zenml.logger import get_logger
+from zenml.models import ServiceConnectorRequirements
+from zenml.orchestrators import BaseOrchestratorConfig, BaseOrchestratorFlavor
+
+if TYPE_CHECKING:
+    from zenml.integrations.skypilot.orchestrators import (
+        SkypilotAWSOrchestrator,
+    )
+
+
+logger = get_logger(__name__)
+
+
+class SkypilotAWSOrchestratorSettings(SkypilotBaseOrchestratorSettings):
+    """Skypilot orchestrator settings."""
+
+
+class SkypilotAWSOrchestratorConfig(  # type: ignore[misc] # https://github.com/pydantic/pydantic/issues/4173
+    SkypilotBaseOrchestratorConfig, SkypilotAWSOrchestratorSettings
+):
+    """Skypilot orchestrator config."""
+
+    @property
+    def is_local(self) -> bool:
+        """Checks if this stack component is running locally.
+
+        This designation is used to determine if the stack component can be
+        shared with other users or if it is only usable on the local host.
+
+        Returns:
+            True if this config is for a local component, False otherwise.
+        """
+        return False
+
+
+class SkypilotAWSOrchestratorFlavor(BaseOrchestratorFlavor):
+    """Flavor for the Skypilot orchestrator."""
+
+    @property
+    def name(self) -> str:
+        """Name of the orchestrator flavor.
+
+        Returns:
+            Name of the orchestrator flavor.
+        """
+        return SKYPILOT_AWS_ORCHESTRATOR_FLAVOR
+
+    @property
+    def service_connector_requirements(
+        self,
+    ) -> Optional[ServiceConnectorRequirements]:
+        """Service connector resource requirements for service connectors.
+
+        Specifies resource requirements that are used to filter the available
+        service connector types that are compatible with this flavor.
+
+        Returns:
+            Requirements for compatible service connectors, if a service
+            connector is required for this flavor.
+        """
+        return ServiceConnectorRequirements(
+            resource_type="aws-generic",
+        )
+
+    @property
+    def docs_url(self) -> Optional[str]:
+        """A url to point at docs explaining this flavor.
+
+        Returns:
+            A flavor docs url.
+        """
+        return self.generate_default_docs_url()
+
+    @property
+    def sdk_docs_url(self) -> Optional[str]:
+        """A url to point at SDK docs explaining this flavor.
+
+        Returns:
+            A flavor SDK docs url.
+        """
+        return self.generate_default_sdk_docs_url()
+
+    @property
+    def logo_url(self) -> str:
+        """A url to represent the flavor in the dashboard.
+
+        Returns:
+            The flavor logo.
+        """
+        return "https://public-flavor-logos.s3.eu-central-1.amazonaws.com/container_registry/aws.png"
```

Comment:
This uses the ECR logo, wouldn't we rather want to use some SkyPilot logo here?

Reply:
Hmm this is the AWS flavor of it, if we just used skypilot one would be hard to differentiate between cloud flavors of it

Reply:
Maybe we could make three simple custom logos with Skypilot logo + AWS/GCP/Azure small logo in a corner or sth?

Reply:
@fa9r that would make sense only if we merge the flavors into one flavor which is currently blocked by the fact that only one service connector can be registered at a time with one component

Reply:
Let's move all further discussions to the GCP comment